grakn-graql/src/main/java/ai/grakn/graql/internal/query/UndefineQueryImpl.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.graql.internal.query;
+
+import ai.grakn.GraknTx;
+import ai.grakn.graql.Printer;
+import ai.grakn.graql.Query;
+import ai.grakn.graql.UndefineQuery;
+import ai.grakn.graql.admin.VarPatternAdmin;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+import java.util.stream.Stream;
+
+import static ai.grakn.util.CommonUtil.toImmutableList;
+
+/**
+ * @author Felix Chapman
+ */
+@AutoValue
+abstract class UndefineQueryImpl implements UndefineQuery {
```

Comment:
What's with all the nulls?

Reply:
haha

it's been a long day

will fix